node_exporter_test.go

```diff
@@ -129,14 +129,12 @@ func runCommandAndTests(cmd *exec.Cmd, address string, fn func(pid int) error) e
 		errc <- fn(pid)
 	}(cmd.Process.Pid)
 
-	select {
-	case err := <-errc:
-		if cmd.Process != nil {
-			cmd.Process.Kill()
-		}
-		if err != nil {
-			return err
-		}
+	err := <-errc
+	if cmd.Process != nil {
+		cmd.Process.Kill()
+	}
+	if err != nil {
```

Comment:
Well, if we're at this, no need for this if statement: `return err`.

Reply:
Hah, yeah, or rather no need for the `return nil` right?

Reply:
No, it's easier to just drop the `if` and return `err`